configBase.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const axios = require('axios')
+const FormData = require('form-data')
+const fs = require('fs')
+const path = require('path')
+
+const SERVICE_NAME = process.env.SERVICE_NAME
+const API_PASSWORD = process.env.API_PASSWORD
+const SERVICE_API_PORT = process.env.SERVICE_API_PORT
+const API_PATH = process.env.API_PATH
+const API_USERNAME = process.env.API_USERNAME
+const JS_REPORT_SSL = process.env.JS_REPORT_SSL
+const BODY_DATA_FORMAT = process.env.BODY_DATA_FORMAT
+
+let data
+switch (BODY_DATA_FORMAT) {
+  case 'multipart/form-data':
+    data = new FormData()
+    data.append('form', fs.createReadStream(path.resolve(__dirname, 'export.jsrexport')))
+    break;
+
+  case 'JSON':
+    const jsonData = JSON.parse(
+      fs.readFileSync(path.resolve(__dirname, 'openhim-import.json'))
+    )
+    data = JSON.stringify(jsonData)
+    break;
+
+  default:
+    break;
+}
+
+const protocol = JS_REPORT_SSL == 'false' ? 'http' : 'https'
+
+let config = {
+  method: 'post',
+  url: `${protocol}://${SERVICE_NAME}:${SERVICE_API_PORT}${API_PATH}`,
+  auth: {
+    username: API_USERNAME,
+    password: API_PASSWORD
+  },
+  headers: {
+    'Content-Type': 'application/json',
+    ...data.getHeaders()
```

Comment:
This will fail if data is a stringified result, perhaps you could cater for this with declaring the dataheaders separately with a default and then speading it afterwards
```
const dataHeaders = data?.getHeaders() ?? {}
```
```
...dataHeaders
```

Reply:
Good suggestion, fixed in [a200f8b](https://github.com/jembi/api-config-importer/pull/1/commits/a200f8b988ed6002aac8af21f19c2776953784f8)